commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt

```diff
@@ -47,26 +50,23 @@ internal class TsvChunkGenerator(private val parser: CSVFormat = createParser())
     }
 
     private fun chunkLines(pageTab: String): List<TsvChunkLine> {
-        val parsedChunks = parser.parse(StringReader(pageTab))
-
-        return parsedChunks.mapIndexed { idx, csvRecord ->
-            val record = csvRecord.asList()
-            when {
-                record.all(String::isBlank) -> TsvChunkLine(idx, emptyList())
-                else -> TsvChunkLine(idx, record)
-            }
-        }
+        val parsedChunks = parser.parseAll(StringReader(pageTab))
+        return parsedChunks.mapIndexed(this::asTsvChunkLine)
     }
 
-    private fun CSVRecord.asList(): List<String> = map { it }
+    private fun asTsvChunkLine(idx: Int, values: Array<String?>): TsvChunkLine =
+        if (values.all { it.isNullOrBlank() }) TsvChunkLine(idx) else TsvChunkLine(idx, asList(values))
+
+    private fun asList(values: Array<String?>) = values.map { it.orEmpty() }.toList()
 
     companion object {
-        private fun createParser(): CSVFormat {
-            return CSVFormat.DEFAULT
-                .withDelimiter(TAB)
-                .withIgnoreSurroundingSpaces()
-                .withIgnoreEmptyLines(false)
-                .withCommentMarker(TSV_COMMENT)
+        private fun createParser(): CsvParser {
+            val settings = CsvParserSettings()
+            settings.format.setLineSeparator("\n")
```

Comment:
Is this the line that fixes the bug? I didn't see any other difference compared to the previous config. If son, couldn't this be added to the existing config instead of changing the whole block? I think the builder approach is clearer

Reply:
added documentation